### ISSUE TITLE
Inconsistent Behavior When Using tf.keras.metrics.Accuracy with F1-Score and Precision

### ISSUE THREAD

**Issue by JacksonDivakar** [tensorflow/tensorflow#87893]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.18.0

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 24.0

### Python version

3.12

### Current behavior?

![Image](https://github.com/user-attachments/assets/9401b0ee-94d5-425a-a152-c8f10629cb82)
![Image](https://github.com/user-attachments/assets/901cedba-add4-43ba-8d2a-963e52046289)
![Image](https://github.com/user-attachments/assets/5efe07d3-9ff8-440d-ad25-c3a6bb0e01e9)

### Standalone code to reproduce the issue

```shell
1)Compile a model using tf.keras.metrics.Accuracy() along with custom F1-score and precision metrics.
2)Observe the incorrect behavior in the reported accuracy.
3)Compare with behavior when using "accuracy" in the metrics list.

Link : https://github.com/JacksonDivakarProjects/Others/blob/main/accuracy_issue%20(2).ipynb
```

### Relevant log output

```shell

```

**Comment by Venkat6871** [comment]
Hi **@JacksonDivakar** ,
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues) as this issue is more related to keras
Thank you!
